pkg/csi/service/node.go

```diff
@@ -385,7 +385,12 @@ func (driver *vsphereCSIDriver) NodeGetInfo(
 		accessibleTopology map[string]string
 	)
 
-	if cnstypes.CnsClusterFlavor(os.Getenv(csitypes.EnvClusterFlavor)) == cnstypes.CnsClusterFlavorGuest {
+	clusterFlavor, err = cnsconfig.GetClusterFlavor(ctx)
```

Comment:
No no. This is not a valid fix. 
You have make sure CLUSTER_FLAVOR is set for vanilla with this change in YAML files. 
Currently we do not have CLUSTER_FLAVOR in vanilla YAML files.